pkg/hooks/hooks.go

```diff
@@ -94,29 +96,83 @@ func runDirectory(dirpath string, environment []string, logger logging.Logger) e
 		if f.IsDir() {
 			continue
 		}
-		logger.WithField("path", fullpath).Infof("Executing hook %s", f.Name())
-		cmd := exec.Command(fullpath)
-		hookOut := &bytes.Buffer{}
-		cmd.Stdout = hookOut
-		cmd.Stderr = hookOut
-		cmd.Env = environment
-		err := cmd.Run()
-		if err != nil {
-			logger.WithErrorAndFields(err, logrus.Fields{
-				"path":   fullpath,
-				"output": hookOut.String(),
-			}).Warnf("Could not execute hook %s", f.Name())
-		} else {
-			logger.WithFields(logrus.Fields{
-				"path":   fullpath,
-				"output": hookOut.String(),
-			}).Debugln("Executed hook")
+
+		timeout := 5
+		h := Hook{fullpath, f.Name(), environment, logger, timeout}
+
+		err := h.RunWithTimeout()
+		htErr, ok := err.(HookTimeoutError)
+		if !ok {
```

Comment:
I'm pretty sure this isn't correct - if `!ok`, then `htErr` is `nil` because the type assertion will have failed. More likely you want to enter this condition `if ok {`. You'll also want to verify that `err` is not nil.